docs/extensibility/custom-non-container-resource.md

```diff
@@ -0,0 +1,145 @@
+- **Configuration values**: For simple configuration, <xref:Aspire.Hosting.ParameterResourceBuilderExtensions.AddParameter*> might be enough.
+
+Custom non-container resources are valuable when you need:
+
+- Dashboard integration with status updates and logs
+- Lifecycle management (starting/stopping services)
+- In-process services that benefit from Aspire's orchestration
+- External resource management with rich feedback
+
+Examples include:
+
+- In-process HTTP proxies or middleware
+- Cloud service provisioning and management
+- External API integrations with health monitoring
+- Background services that need dashboard visibility
+
+## Key components
+
+Non-container custom resources use these key Aspire services:
+
+- **<xref:Aspire.Hosting.Lifecycle.IDistributedApplicationLifecycleHook>**: Hook into app startup/shutdown. For more information, see [App Host life cycle events](../app-host/eventing.md#app-host-life-cycle-events).
+- **<xref:Microsoft.Extensions.Logging.ILogger>**: Standard .NET logging that appears in console and dashboard
+
+> [!NOTE]
+> Advanced dashboard integration is possible using services like `ResourceNotificationService` and `ResourceLoggerService` for real-time status updates and log streaming. These APIs provide richer dashboard experiences but require more complex implementation.
+
+## Example: HTTP proxy resource
+
+This example creates an in-process HTTP proxy resource that demonstrates the core concepts of lifecycle management and logging integration with the Aspire dashboard.
+
+### Define the resource
+
+First, create the resource class:
+
+:::code language="csharp" source="snippets/HttpProxyResource/HttpProxy.Hosting/HttpProxyResource.cs":::
+
+The resource implements <xref:Aspire.Hosting.ApplicationModel.IResource> and includes properties for the proxy configuration.
+
+### Create the extension method
+
+Next, create the builder extension:
+
+:::code language="csharp" source="snippets/HttpProxyResource/HttpProxy.Hosting/HttpProxyResourceBuilderExtensions.cs":::
+
+This extension method adds the resource to the application model and configures an HTTP endpoint.
+
+### Implement lifecycle management
+
+Create a lifecycle hook to manage the proxy:
+
+:::code language="csharp" source="snippets/HttpProxyResource/HttpProxy.Hosting/HttpProxyLifecycleHook.cs":::
+
+The lifecycle hook:
+
+1. **Manages lifecycle**: Starts services when resources are created
+2. **Integrates logging**: Uses standard .NET logging that appears in the Aspire dashboard
+3. **Handles background tasks**: Runs long-running services in background tasks
+4. **Provides resource management**: Manages resources like HTTP listeners and cleanup
+
+### Register the lifecycle hook
+
+The extension method automatically registers the lifecycle hook:
+
+```csharp
+public static IResourceBuilder<HttpProxyResource> AddHttpProxy(
```

Comment:
This code block should instead use the `:::code` syntax instead to pull in the _HttpProxyResourceBuilderExtensions.cs_ file.